### ISSUE TITLE
Exclude properties from update in BulkInsertOrUpdate method

### ISSUE THREAD

**Issue by jarszkutn** [borisdj/EFCore.BulkExtensions#155]
I cannot find an option to exclude properties from the update in BulkInsertOrUpdate method.
Let's say we have CreationDate/CreationBy property - it should be inserted but never updated.

I've added the property to Exclusion list but it does not work for new recrods - sql exception received (null cannot be inserted to this column)

**Comment by borisdj** [comment]
When Insert and/or Update is done together with one BulkInsertOrUpdate method then 
PropertiesToExclude is shared.
If exclusion list are different for Insert vs Update then you should use 2 lists and call separately BulkInsert and then  BulkUpdate.

**Comment by jarszkutn** [comment]
Yes, it is kind of workaround, but then I need one more call to get all existing records.. some it has some impact on performance.
Do you plan to do some enhancement to support exclusion only for an update?

**Comment by borisdj** [comment]
True, but I think it can be done in 2 calls.
Because after your read data just set new ones and for update change certain columns while leaving other columns with original values from Db. Then you could use BulkInsertOrUpdate.
Also Reading is mush faster operation there is also BulkRead method so you read only ones you need.
I will consider the enhancement.

**Comment by jwisener** [comment]
:-(

**Comment by rejihcss** [comment]
:-( This is a limitation for us from using this library, as is.

**Comment by dafunkphenomenon** [comment]
This addition would be great for me, I have tons of records coming from an external system that I have to integrate, currently not being able to stop updates on date fields (created) in particular is causing me grief.

Thanks in advance

Matt

**Comment by john10e** [comment]
This feature would be really helpful for similar use case that I've run into. Thanks!

**Comment by codecooper** [comment]
I actually have a solution that I have been using a local fork for the past couple of years. In short, I updated the section that writes data to the final table and tell it to check only certain columns for changes. I am a data warehouse architect and use this features all the time for delta processing.
For a bit longer of an explanation, I added two new BulkConfig properties **PropertiesToExcludeOnCompare** and **PropertiesToIncludeOnCompare**. These properties behave just like **PropertiesToExclude** and **PropertiesToInclude** except they are used to ignore or explicitly include columns. I then modified the MERGE statement to add a new condition on *WHEN MATCHED* that will only allow an update if and of the allowed columns are different. Everything else behaves as normal (such as updating the CreationDate/CreationBy and UpdateDate/UpdateBy columns)
I actually came here to add these changes, finally, to the project.

**Comment by codecooper** [comment]
I just realized that by response above was for update fields and not create. While I will still get the update part added, the create timestamp is a bit more straight-forward. In the past, I have just added a default constraint to the timestamp column then add that column to **PropertiesToExclude**.

**Comment by borisdj** [comment]
Closing the issue since @codecooper PR is merged.
Great work.

**Comment by Brondahl** [comment]
Hi, It looks like that PR only applies to `SqlQueryBuilder` but not to `PostgresSqlQueryBuilder`.
If you set that OnCompare config for a PostGres connection, it just doesn't work at all.

~I couldn't find anything documenting this; I only established it through a fairly painful testing process and stepping through the codebase.~ I failed to see the line in the README that documented this, and thus wasted several extra hours 🤦‍♂️ 

Is it possible to extend the functionality to the Postgres integration too?
If not, then IMO this should have an error trigger to throw if incompatible config is provided :(
